### PR TITLE
Better Configuration UX

### PR DESCRIPTION
**Pull request by Christian7573**
I plan on embedding wetty in a project of mine and looked into changing the color theme. While the previous editor allowed changing the color theme, it was very clunky to use, needing to ready XTerm documentation to find property names and then painfully edit a tiny JSON window on my phone. I've replaced it with a simple field-value editor and copied over all of the options listed in XTerm's documentation. It looks like you have this project set up in a very specific way, so I've done my best to keep it isolated in the assets folder using vanilla HTML, CSS, and JS.

image

- Replaced the JSON editor with a field-value GUI.
- Modified the loading, saving, and application of XTerm configuration.
- Old options strings will be converted to { xterm: old_options } upon using the new version.
- Added the option to disable the XTerm Fit plugin if people choose to manually specify rows and columns.
- Changed sizing of configuration editor to better work with smaller and mobile screens.
- Bumped version of @typescript/eslint-plugin; NPM could not resolve the version previously listed in package.json.
- Added package-lock.json and yarn.lock to the .gitignore.
- Expose the Term object as window.wetty_term; Not related to this PR, but I'll be using it to access the terminal externally for my project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,8 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
+/package-lock.json
+/yarn.lock
```
Comment (butlerx) [inline .gitignore]: Lock files shouldn't be in the git ignore

**Review by butlerx — CHANGES_REQUESTED** [review]

**Comment by Christian7573** [comment]
Interesting to note: typescript/eslint-plugin 2.5.0 is no longer available from npm and only available through yarn.
Let me know if you'd like me to squash commits.

File: src/client/wetty/term/confiruragtion/load.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import _ from 'lodash';
 
-export function loadOptions(): object {
-  const defaultOptions = { fontSize: 14 };
+export function loadOptions(): any {
```
Comment (butlerx) [inline src/client/wetty/term/confiruragtion/load.ts]: Why switch from `object` to `any`?

**Review by butlerx — COMMENTED** [review]
I'm curious to know the reasoning behind using an iframe rather than adding to the main page?

File: src/client/wetty/term/confiruragtion/load.ts (hunk 2 above)
Comment (Christian7573) [reply src/client/wetty/term/confiruragtion/load.ts]: I believe TS didn't like me indexing/using properties from type object, however changing it back to object now does not bring that error up. Will revert.

**Comment by Christian7573** [comment]
> I'm curious to know the reasoning behind using an iframe rather than adding to the main page?

To have a hard boundary for CSS and JS since it's neither using SCSS nor bundled TS.

**Pull request merged**